app/controllers/posts_controller.rb

```diff
@@ -1,20 +1,12 @@
 class PostsController < ApplicationController
   def index
-    @current_user = current_user
     @user = User.find(params[:user_id])
-
-    redirect_to root_path if @user.nil?
-    @posts = @user.posts
+    @posts = Post.includes(:comments).where(author_id: @user.id)
   end
 
   def show
     @user = User.find(params[:user_id])
-
-    redirect_to root_path if @user.nil?
-    @post = @user.posts.find(params[:id])
-
-    redirect_to user_posts_path(@user) if @post.nil?
-    @comments = @post.comments
+    @post = Post.includes(:comments).find_by(id: params[:id])
   end
```

Comment:
![image](https://github.com/Microverse-Fullstack-Program/rails-blog-app-v2/assets/93039359/82a753be-eff7-4524-915c-1e46e20a6164)

- [ ] Great work on fixing the N+1 problem 👍 but it would be great if you provide us the screenshots of the logs before and after fixing the N+1 problem and put them in the pull request summary so we can verify there are no N+1 issues in your app anymore 😊

**NOTE: I can see you have provided some screenshots but dear you need to add the screenshots of the logs, not the code itself 😄 If you face any trouble then reach out to me on slack without any hesitation 👍 You can reach out to me by searching my name in the full-time-students channel 😊**